components/motor/gpio/controlled.go

```diff
@@ -150,6 +155,8 @@ type controlledMotor struct {
 	controlLoopConfig control.Config
 	blockNames        map[string][]string
 	loop              *control.Loop
+	configPIDVals     []control.PIDConfig
+	tunedVals         *[]control.PIDConfig
```

Comment:
The question that pops up in my mind while reading the variables in this struct is - we're already storing the whole config, and constructing a new stored variable from the PID parameters we see in the control config. Are we accruing too many variables for maintainable code? Can we cut out some? This was the type of coding that resulted in  made encoded_motor's old state. JUst calling it out, john might tackle this in his refactor.

Reply:
I can remove the config pid values. just seemed easier than digging through the config but I can make a helper for it. can potentially remove tunedVals as well but I need to double check that they are being stored in the config once tuned

Reply:
i actually tried removing tunedVals before when I was testing something else in this code, the issue I ran into was there wasn't actually anything else being stored in our state that contained the tunedVals. 

we could maybe store the [PIDLoop](https://github.com/viamrobotics/rdk/blob/98d78c38fdee44edb60d145ff661c6896a2b7baf/control/setup_control.go#L36C6-L36C14) struct instead of breaking out the different values in the config, but I kinda agree we can think about this cleanup abit more later

Reply:
thought about that as well but it's kind of a lot. I think we have to keep one. we can either store the tuned values and get the config values from the config every time we need them, or we can store the config values, update the config with the tuned values, and then get the tuned values from the config every time we need them so I personally think it's better to not edit the config during the tuning process.

Reply:
actually I take it back I think it is best to store both, because otherwise we have to store the block indices for the linear and angular PID blocks which are a part of the setup functions, so they either have to be stored also as a part of the sensor controlled base, or exported publicly from the setup control file which is worse imo. I am sure there's a better way to set this up that it isn't all necessary, but I think maybe for this pass it's the best option? not 100% sure, let me know what you think or if you have better ideas

Reply:
OR instead of storing anything we can just loop through the entire config and find the values. I don't like doing that personally when they could be stored at the beginning when we are accessing the config during reconfigure but maybe you both feel differently

Reply:
^ we should do an in-person  review of the control helpers and this code with John, because right now, it's a lot of setup from the control package, I do want to develop it into a better library with more intuitive functions with less overhead. Next pr.